### DISCUSSION TITLE
lotus-miner failed to initialize。show  Waiting for confirmation

### DISCUSSION THREAD

**Discussion by linyintao2010** [filecoin-project/lotus#8939] [General Q&A]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [ ] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [X] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [X] Other

### Lotus Version

```text
Daemon:  1.17.0-rc1+2k+git.d2fe153e3+api1.5.0
Local: lotus version 1.17.0-rc1+2k+git.d2fe153e
```

### Describe the Bug

lotus-miner failed to initialize。show  Waiting for confirmation。Stuck without moving

try to exec  ./lotus mpool replace -auto <from> <none> not work。

### Logging Information

```text
2022-06-28T15:33:34.070+0800	INFO	paramfetch	go-paramfetch@v0.0.4/paramfetch.go:209	Parameter file /var/tmp/filecoin-proof-parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-0-0-sha256_hasher-032d3138d22506ec0082ed72b2dcba18df18477904e35bafee82b3793b06832f.params is ok
2022-06-28T15:33:34.070+0800	INFO	paramfetch	go-paramfetch@v0.0.4/paramfetch.go:233	parameter and key-fetching complete
2022-06-28T15:33:34.070+0800	INFO	main	lotus-miner/init.go:157	Trying to connect to full node RPC
2022-06-28T15:33:34.073+0800	INFO	main	lotus-miner/init.go:169	Checking full node sync status
2022-06-28T15:33:34.073+0800	INFO	main	lotus-miner/init.go:177	Checking if repo exists
2022-06-28T15:33:34.073+0800	INFO	main	lotus-miner/init.go:193	Checking full node version
2022-06-28T15:33:34.073+0800	INFO	main	lotus-miner/init.go:204	Initializing repo
2022-06-28T15:33:34.073+0800	INFO	repo	repo/fsrepo.go:265	Initializing repo at '/home/ubuntu/.lotus-miner-local-net'
2022-06-28T15:33:34.074+0800	INFO	main	lotus-miner/init.go:419	Initializing libp2p identity
2022-06-28T15:33:34.091+0800	INFO	badger	v2@v2.2007.3/levels.go:183	All 0 tables opened in 0s

2022-06-28T15:33:34.121+0800	INFO	badger	v2@v2.2007.3/levels.go:183	All 0 tables opened in 0s

2022-06-28T15:33:34.207+0800	INFO	main	lotus-miner/init.go:748	Pushed CreateMiner message: bafy2bzacedg3vbyfcgfhwq2lkrafvhjqmiemscpyqjapguugipybtrk42l2fm
2022-06-28T15:33:34.207+0800	INFO	main	lotus-miner/init.go:749	Waiting for confirmation
```

### Repo Steps

1. lotus-miner init --sector-size=2KiB --nosync

2.try to exec  ./lotus mpool replace -auto <from> <none> not work。

3. lotus daemon Synchronizing block data。why can not Broadcast the message pool

**Comment by rjan90** [comment]
Hey @linyintao2010! It seems like you are trying to initialize on a 2k/developer network. Is this network up and running, and are you connected to it?

I also do not reccomend trying to initilize the storage provider with the `--nosync` flag

**Reply by linyintao2010** [reply]
> Hey @linyintao2010! It seems like you are trying to initialize on a 2k/developer network. Is this network up and running, and are you connected to it?
> 
> I also do not reccomend trying to initilize the storage provider with the `--nosync` flag

can connected it  and  see the synchronization block，The height is increasing all the time。According to my understanding of blockchain, as long as the blocks are synchronized, everything is fine，But filecoin the synchronous wait is never done in the file currency，miner init always show wait confirmation。

I noticed that diff 1 appeared when I executed the following command (lotus sync wait）

**Comment by rjan90** [comment]
Did you follow [this guide](https://lotus.filecoin.io/developers/local-network/) setting up the local developer network?

I´m also converting this issue into a discussion, as it this does not currently seem like an issue, but rather a troubleshooting question.

**Reply by linyintao2010** [reply]
> Did you follow [this guide](https://lotus.filecoin.io/developers/local-network/) setting up the local developer network?
> 
> I´m also converting this issue into a discussion, as it this does not currently seem like an issue, but rather a troubleshooting question.

yes,strictly follow the local network guide. Even at genesis, the first miner's servers failed to synchronize，Meanwhile, I can see that blocks are being created all the time，That means the miners are working。Execute the following command： Lotus Sync Wait on this server does not see done。
Everything looks fine, why can't it be done。Are there any more Settings to output more information to find the problem，thanks

**Comment by LexLuthr** [comment]
Hey @linyintao2010 , Are you sure that you have initialised the miner correctly?
`./lotus-miner init --genesis-miner --actor=t01000 --sector-size=2KiB --pre-sealed-sectors=~/.genesis-sectors --pre-sealed-metadata=~/.genesis-sectors/pre-seal-t01000.json --nosync`

If you will try to initialise the miner without 2 sectors sealed and specifying the genesis miner ID, it will not work. The reason is simple, a message(miner init message) requires to be included in the block and there are no miners in local-network to create blocks and execute messages. It is chicken-egg conundrum. So we init the miner using above command that includes all the required information to initialise the miner.

**Reply by linyintao2010** [reply]
> Hey @linyintao2010 , Are you sure that you have initialised the miner correctly? `./lotus-miner init --genesis-miner --actor=t01000 --sector-size=2KiB --pre-sealed-sectors=~/.genesis-sectors --pre-sealed-metadata=~/.genesis-sectors/pre-seal-t01000.json --nosync`
> 
> If you will try to initialise the miner without 2 sectors sealed and specifying the genesis miner ID, it will not work. The reason is simple, a message(miner init message) requires to be included in the block and there are no miners in local-network to create blocks and execute messages. It is chicken-egg conundrum. So we init the miner using above command that includes all the required information to initialise the miner.

yes，I can see that height is increasing all the time。At the same time, I tried to transfer money, and I could see the new account received

**Comment by linyintao2010** [comment]
Hey @rjan90 ，can you help me solve this problem。

I follow [this guide](https://lotus.filecoin.io/developers/local-network/) setting up the local developer network in new server。
Create the initial node and miner，I can see that height is increasing all the time。
then execute the following command
./lotus sync wait
You can see that validates a lot of information，and diff is zero，see below img
![image](https://user-images.githubusercontent.com/30648802/176583144-01920f9d-bb2e-4a2a-b815-58a4e8ed168e.png)

Results of the above execution What puzzles me is why it's not done，Is this normal？

How can I verify the success of setting up the local network in filecoin。

thanks

**Reply by LexLuthr** [reply]
This is expected behaviour as there is nothing to sync from. You are the only node in this network as well as the only block producer. 
You can dig through the command code to understand why it never completes. 

If you are able to send funds then network is setup correctly.

**Comment by linyintao2010** [comment]
> lotus-miner failed to initialize。show Waiting for confirmation。Stuck without moving
> 
> try to exec ./lotus mpool replace -auto not work。



hey @LexLuthr 
in the second server  the second lotus daemon success to initialize,can see sync height. but the second lotus-miner failed to initialize。show Waiting for confirmation。Stuck without moving

try to exec ./lotus mpool replace -auto not work。

follow [this guide](https://github.com/filecoin-project/community-china/blob/master/documents/tutorial/local_2k_dev_tutorial/local_2k_dev_tutorial.md/) setting up the local developer network.

how to go next step to solve this problem. thanks

**Comment by TippyFlitsUK** [comment]
Hi @linyintao2010 

Are you still experiencing this issue?
